[None][feat] Adding torch ext API for FusedAddRMSNormQuant kernel

JintaoPengCS opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a new fused kernel operation combining addition, RMSNorm/LayerNorm, and FP4 quantization in a single step. Available through both C++ and Python interfaces with support for fp16 and bf16 data types on compatible GPU architectures.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

<!--
Please explain the issue and the solution in short.
-->

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

JintaoPengCS commented:
/bot run --disable-fail-fast

JintaoPengCS commented:
/bot run --disable-fail-fast

zongfeijing commented:
/bot run --disable-fail-fast

JintaoPengCS commented:
/bot run --disable-fail-fast

benzh-2025 commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #31234](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31234/) [ run ] completed with state `SUCCESS`. Commit: `8e9e31d` 
[/LLM/main/L0_MergeRequest_PR pipeline #24138](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24138/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

dc3671 commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #31434](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31434/) [ run ] triggered by Bot. Commit: `8e9e31d`

tensorrt-cicd commented:
[PR_Github #31434](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31434/) [ run ] completed with state `SUCCESS`. Commit: `8e9e31d` 
[/LLM/main/L0_MergeRequest_PR pipeline #24295](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24295/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

yizhang-nv commented on tensorrt_llm/_torch/models/modeling_llama.py:
Please remove the comment here.

HuiGao-NV reviewed (approved):
LGTM

tensorrt-cicd commented:
[PR_Github #31559](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31559/) [ run ] completed with state `SUCCESS`. Commit: `451dcef` 
[/LLM/main/L0_MergeRequest_PR pipeline #24402](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24402/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

JintaoPengCS committed:
update torch_ext API and debugging test for FusedAddRMSNorm

update #define for hopper & blackwell

Update cpp/tensorrt_llm/thop/fusedAddRMSNormQuant.cpp

Signed-off-by: jintaop <jintaop@nvidia.com>

zongfeijing commented:
/bot run --disable-fail-fast

JintaoPengCS committed:
Merge upstream/main to sync with NVIDIA/TensorRT-LLM

Signed-off-by: jintaop <jintaop@nvidia.com>

zongfeijing commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #31842](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31842/) [ run ] completed with state `SUCCESS`. Commit: `5f42be3` 
[/LLM/main/L0_MergeRequest_PR pipeline #24655](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24655/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

JintaoPengCS committed:
Merge upstream/main

JintaoPengCS committed:
Merge upstream/main

JintaoPengCS committed:
Update llama torch model + llm api pytorch test

Signed-off-by: jintaop <jintaop@nvidia.com>

JintaoPengCS committed:
Update llm api pytorch accuracy test

Signed-off-by: jintaop <jintaop@nvidia.com>

tensorrt-cicd commented:
[PR_Github #31929](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31929/) [ kill ] triggered by Bot. Commit: `5feb18a`

JintaoPengCS committed:
Merge branch 'NVIDIA:main' into main
